osdep/timer-win32.c

```diff
@@ -27,37 +29,91 @@
 
 static LARGE_INTEGER perf_freq;
 
-// ms values
-static int hires_max = 50;
-static int hires_res = 1;
+// ns values
+static int64_t hires_max = MP_TIME_US_TO_NS(50000);
+static int64_t hires_res = MP_TIME_US_TO_NS(1000);
 
-int mp_start_hires_timers(int wait_ms)
+// NtSetTimerResolution allows setting the timer resolution to less than 1 ms.
+// Resolutions are specified in 100-ns units.
+// If Set is TRUE, set the RequestedResolution. Otherwise, return to the previous resolution.
+static NTSTATUS (NTAPI *pNtSetTimerResolution)(ULONG RequestedResolution, BOOLEAN Set, PULONG ActualResolution);
+// Acquire the valid timer resolution range.
+static NTSTATUS (NTAPI *pNtQueryTimerResolution)(PULONG MinimumResolution, PULONG MaximumResolution, PULONG ActualResolution);
```

Comment:
Please go back to your previous version. It works a links correctly.

You are missing calling convention attribute. You have to add `NTAPI` and it will like correctly, only 32-bit build were affected.

`NTSTATUS NTAPI NtSetTimerResolution(...)`

Also like mentioned before test/meson.build is missing `ntdll`.

And please remove `NtQueryTimerResolution` it is already defined.

Reply:
>  NtQueryTimerResolution() is included for msys2 but not mingw so it still need to be included.

I see, ok.